src/main/java/com/intuit/graphql/orchestrator/federation/EntityQuery.java

```diff
@@ -0,0 +1,93 @@
+package com.intuit.graphql.orchestrator.federation;
+
+import static graphql.language.AstPrinter.printAstCompact;
+import static graphql.language.TypeName.newTypeName;
+
+import graphql.ExecutionInput;
+import graphql.GraphQLContext;
+import graphql.language.Argument;
+import graphql.language.Document;
+import graphql.language.Field;
+import graphql.language.InlineFragment;
+import graphql.language.OperationDefinition;
+import graphql.language.OperationDefinition.Operation;
+import graphql.language.Selection;
+import graphql.language.SelectionSet;
+import graphql.language.VariableDefinition;
+import graphql.language.VariableReference;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import lombok.Builder;
+import org.apache.commons.collections4.CollectionUtils;
+
+/** This class is used to create a downstream query to make an entity fetch. */
+@Builder
+public class EntityQuery {
+
+  private static final String REPRESENTATIONS_VAR_NAME = "REPRESENTATIONS";
```

Comment:
Should be moved to a FederationConstants

Reply:
Should be in the next external/refactor pr